inference/core/workflows/core_steps/common/utils.py

```diff
@@ -48,38 +50,72 @@ def attach_prediction_type_info(
     return predictions
 
 
-def attach_parent_info(
-    images: List[Dict[str, Any]],
+def convert_to_sv_detections(
```

Comment:
Maybe I do not get the full context here, but if we are converting inference results to sv detections, then maybe we could get rid of parametrisation of fields names that are related to responses in favour of constants (or even get rid of operations on dicts in favour of operating on responses objects, but then what with QR codes and barcodes). Then we could also name the input parameter as `batch_inference_results` and just return `list[sv.Detections]` without embedding that into `dict["predictions"] = [<sv.Detections>]`. That would shift the responsibility of forming proper outputs of blocks into caller which owns this. Obviously that would cost another round of for-loop on the caller side, but what is visible in barcode detection - the fact that other keys that `predictions_key` are forgotten in output list of dicts actually makes this function harder to read and understand in global context.

Reply:
[requires change] this is not for this function directly, but in topic of `sv.Detections` - we need to be able to add metadata to created object - namely predictions type

Intent is the following - all metadata that are directly related to prediction should be packed into sv.Detections to limit the # of auxiliary inter-blocks connections